src/lib.rs

```diff
@@ -50,9 +50,18 @@ pub enum Error {
 pub type Result<T> = result::Result<T, Error>;
 
 /// Opaque object that uniquely represents a subscriber registered with an `EventManager`.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Hash)]
```

Comment:
re-adding the comment here so we can discuss it in this PR instead: https://github.com/rust-vmm/event-manager/pull/27#discussion_r450735240

I still think it is weird for the default to be something invalid. Can you please explain (again) why is default needed?

Reply:
Let's take virtio device driver as an example: a virtio device may store a SubsriberId of its registered epoll handler.
1) We need a default invalid SubscriberId when creating the device driver instance.
2) We register to event manager and receive a valid SubscriberId in activate(), and save the allocated SubscriberId to the device driver instance.
3) Later we could use the sScriberId to look up the epoll handler.
Another way to support this scenario is to use Option<SubscriberId>, but that's a little inconvenient.

Reply:
It looks like a workaround to me (and not a pretty one). I prefer to stick with rust way of handling this (which is to use an Option I suppose).

Reply:
Sure, I will remove this patch from the MR:)